core/src/main/java/org/infinispan/remoting/inboundhandler/DefaultTopologyRunnable.java

```diff
@@ -0,0 +1,50 @@
+package org.infinispan.remoting.inboundhandler;
+
+import org.infinispan.commands.remote.CacheRpcCommand;
+import org.infinispan.remoting.responses.CacheNotFoundResponse;
+import org.infinispan.remoting.responses.Response;
+
+/**
+ * The default {@link Runnable} for the remote commands receives.
+ * <p>
+ * It checks the command topology and ensures that the topology higher or equal is installed in this node.
+ *
+ * @author Pedro Ruivo
+ * @since 8.0
+ */
+public class DefaultTopologyRunnable extends BaseBlockingRunnable {
+
+   private final TopologyMode topologyMode;
+   private final int commandTopologyId;
+
+   public DefaultTopologyRunnable(BasePerCacheInboundInvocationHandler handler, CacheRpcCommand command, Reply reply,
+                                  TopologyMode topologyMode, int commandTopologyId) {
+      super(handler, command, reply);
+      this.topologyMode = topologyMode;
+      this.commandTopologyId = commandTopologyId;
+   }
+
+   @Override
+   public boolean isReady() {
+      return topologyMode.isReady(handler.getStateTransferLock(), waitTopology());
+   }
+
+   @Override
+   protected Response beforeInvoke() throws Exception {
+      if (0 <= commandTopologyId && commandTopologyId < handler.stateTransferManager.getFirstTopologyAsMember()) {
```

Comment:
Doesn't each invocation handler check `isCommandSentBeforeFirstTopology()` before executing the command?